core/vtk/ttkPersistenceDiagram/ttkPersistenceDiagram.cpp

```diff
@@ -48,139 +35,339 @@ int ttkPersistenceDiagram::FillOutputPortInformation(int port,
       info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkUnstructuredGrid");
       break;
   }
-
   return 1;
 }
 
-int ttkPersistenceDiagram::getScalars(vtkDataSet *input) {
-  vtkPointData *pointData = input->GetPointData();
-
-#ifndef TTK_ENABLE_KAMIKAZE
-  if(!pointData) {
-    cerr << "[ttkPersistenceDiagram] Error : input has no point data." << endl;
-    return -1;
-  }
-#endif
-
-  if(ScalarField.length()) {
-    inputScalars_ = pointData->GetArray(ScalarField.data());
-  } else {
-    inputScalars_ = pointData->GetArray(ScalarFieldId);
-    if(inputScalars_)
-      ScalarField = inputScalars_->GetName();
-  }
-
-#ifndef TTK_ENABLE_KAMIKAZE
-  if(!inputScalars_) {
-    cerr
-      << "[ttkPersistenceDiagram] Error : input scalar field pointer is null."
-      << endl;
-    return -3;
-  }
-#endif
-
-  if(this->GetMTime() < inputScalars_->GetMTime())
```

Comment:
I think that was the bit checking for updates.

Reply:
Mh, I don't think it is sensible to compare the modification time of a filter to the modification time of an array. The filter should only update its output if either its input or parameters change. Here it looks like someone (TTK) is overriding in the background an array and then checks if this occurred. Is this the case?

Reply:
sensible or not, the example is broken.
@tbmasood could you add this check back in somehow to see if that's the issue?

Reply:
Yes, this fixes the issue in timeTracking state file. I have just pushed the change.

The tribute state file still has the issue you pointed out.